oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/endpoint/DefaultAuthorizationCodeTokenResponseClient.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.client.endpoint;
+
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.http.converter.FormHttpMessageConverter;
+import org.springframework.security.oauth2.client.endpoint.converter.OAuth2AuthorizationCodeGrantRequestEntityConverter;
```

Comment:
This is a package tangle because a parent package is accessing a subpackage of itself.

Reply:
i'm wondering if to fix the tangle if we could move `OAuth2AuthorizationCodeGrantRequestEntityConverter` and `OAuth2ClientCredentialsGrantRequestEntityConverter` to `org.springframework.security.oauth2.core.http.converter`

Reply:
`OAuth2AuthorizationCodeGrantRequest` and `OAuth2ClientCredentialsGrantRequest` reside in the `client.endpoint` package so we should keep it in the `client` module.

The 2 options I can think of right now are:
`org.springframework.security.oauth2.client.endpoint` (keep in same package)
`org.springframework.security.oauth2.client.converter`

Reply:
Good point, I think given it is associated with RequestEntity, we should consider a variation of what you proposed. What do you think of `org.springframework.security.oauth2.client.http.converter`

Reply:
Agreed that makes sense....let's go with `org.springframework.security.oauth2.client.http.converter`